tests/unit/test_term_set.py

```diff
@@ -215,3 +217,125 @@ def test_wrapper_extend_error(self):
             data_obj.extend(['bad_data'])
+
+class TestTermSetConfig(TestCase):
+    def setUp(self):
+        if not REQUIREMENTS_INSTALLED:
+            self.skipTest("optional LinkML module is not installed")
+
+    def tearDown(self):
+        unload_termset_config()
+
+    def test_get_loaded_config(self):
+        load_termset_config(config_path='tests/unit/hdmf_config.yaml', return_map=True)
+        loaded_config = get_loaded_config()
+        config = {'namespaces': {'hdmf-common': {'version': '3.12.2',
+                  'data_types': {'VectorData': {'description': 'example_test_term_set.yaml'},
+                                 'VectorIndex': {'data': '...'}}}}}
+
+        self.assertEqual(loaded_config, config)
+
+    def test_get_loaded_config_error(self):
+        with self.assertRaises(ValueError):
+            get_loaded_config()
+
+    def test_config_path(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        self.assertEqual(tc.path, [path])
+
+    def test_get_config(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        self.assertEqual(tc.get_config('VectorData', 'hdmf-common'),
+                                      {'description': 'example_test_term_set.yaml'})
+
+    def test_get_config_namespace_error(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        with self.assertRaises(ValueError):
+            tc.get_config('VectorData', 'hdmf-common11')
+
+    def test_get_config_container_error(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        with self.assertRaises(ValueError):
+            tc.get_config('VectorData11', 'hdmf-common')
+
+    def test_already_loaded_path_error(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        tc = TermSetConfigurator(path=path)
+        with self.assertRaises(ValueError):
+            tc.load_termset_config(config_path=path)
+
+    def test_load_two_unique_configs(self):
+        path = 'tests/unit/hdmf_config.yaml'
+        path2 = 'tests/unit/hdmf_config2.yaml'
+        tc = TermSetConfigurator(path=path)
+        tc.load_termset_config(config_path=path2)
+        self.assertEqual(tc.path, [path, path2])
+
+
+class ExtensionContainer(Container):
+    __fields__ = ("description",)
+
+    def __init__(self, **kwargs):
+        description, namespace, type_map = popargs('description', 'namespace', 'type_map', kwargs)
+        super().__init__(**kwargs)
+        self.namespace = namespace
+        self.type_map = type_map
+        self.description = description
+
+
+class TestTermSetConfigVectorData(TestCase):
+    def setUp(self):
+        if not REQUIREMENTS_INSTALLED:
+            self.skipTest("optional LinkML module is not installed")
+
+        self.tm = load_termset_config(config_path='tests/unit/hdmf_config.yaml', return_map=True)
```

Comment:
It looks like `self.tm` is used solely to initialize `ExtensionContainer`. You have code above that sets `type_map = __TYPE_MAP` in `register_container_class`. Instead of using the special testing-only `return_map=True` functionality, we could decorate the `ExtensionContainer` class with `register_container_class`, which would then set the type map on the class, and this would also more accurately mimic a user's actions.

Reply:
There is no spec for ExtensionContainer. In another PR I can expand our test containers a bit, but I think it is more efficient to build this minimally.